### PR TITLE
[feat]rank model supports sample weight

### PR DESCRIPTION
**Pull request by eric-gecheng**
Adapt rank model loss calculation to support sample weight, especially JRC loss.

### CODE REVIEW

File: tzrec/models/rank_model.py [hunk 1]
```diff
@@ -183,6 +185,9 @@ def _loss_impl(
     ) -> Dict[str, torch.Tensor]:
```
Comment (tiankongdeguiji) [inline tzrec/models/rank_model.py:188]: The sample_weight_name parameter, similar to label_name, needs to be an input for the _loss_impl function. In a multi-task ranking model, it's essential to specify distinct sample weights for each loss function.

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: tzrec/loss/jrc_loss.py [hunk 2]
```diff
@@ -88,8 +102,15 @@ def forward(self, logits: Tensor, labels: Tensor, session_ids: Tensor) -> Tensor
             logits_neg + ((1 - neg_session_mask) + (1 - diag_neg) * y_neg) * -1e9
         )
         loss_neg = self._ce_loss(logits_neg, neg_diag_label)
+        loss_pos = loss_pos * pos_num / batch_size
+        loss_neg = loss_neg * neg_num / batch_size
+        if self._reduction != "none":
+            ge_loss = loss_pos + loss_neg
+        else:
+            ge_loss = torch.zeros_like(labels, dtype=torch.float)
```
Comment (tiankongdeguiji) [inline tzrec/loss/jrc_loss.py:111]: add a test for  reduction=='mean' in jrc_loss_test.py

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: tzrec/models/rank_model.py (hunk 1 above)
Comment (eric-gecheng) [reply tzrec/models/rank_model.py:188]: fixed

File: tzrec/loss/jrc_loss.py (hunk 2 above)
Comment (eric-gecheng) [reply tzrec/loss/jrc_loss.py:111]: added a test for reduction == 'none' since 'mean' is the default option.

**Review by tiankongdeguiji — APPROVED** [review]
LGTM
